src/robot/api/deco.py

```diff
@@ -96,9 +128,38 @@ def decorator(func):
     return decorator
 
 
+C = TypeVar('C')
+
+
+# Decorator used without arguments, i.e. @library()
+@overload
+def library(scope: Type[C]) -> Type[C]:
+    ...
+
+
+# Decorator used with arguments, e.g. @library(version="1.0.0")
+@overload
+def library(
+    scope: 'str | None' = None,
+    version: 'str| None' = None,
+    converters: 'Dict[Any, Callable[..., Any]] | None' = None,
```

Comment:
Converter typing isn't correct. I believe it should be `dict[type, Callable[[Any], Any]]` in the common case. Converter methods have an optional second argument, though, that will be the library instance. I don't know what's the best way to type optional arguments, but most likely this will require a type alias. Alternatively we could use just `dict[type, Callable]` which at least doesn't cause false negatives.